src/allmydata/test/test_util.py

```diff
@@ -546,6 +554,7 @@ class YAML(unittest.TestCase):
     def test_convert(self):
         data = yaml.safe_dump(["str", u"unicode", u"\u1234nicode"])
         back = yamlutil.safe_load(data)
-        self.failUnlessEqual(type(back[0]), unicode)
-        self.failUnlessEqual(type(back[1]), unicode)
-        self.failUnlessEqual(type(back[2]), unicode)
+        str_type = type("x") # workaround for future.builtins.str
+        self.failUnlessEqual(type(back[0]), str_type)
+        self.failUnlessEqual(type(back[1]), str_type)
+        self.failUnlessEqual(type(back[2]), str_type)
```

Comment:
It looks like the yaml module returns unicode instead of bytes for ASCII-only strings on Python 3.6 (unlike its behavior on Python 2.7 where it returns bytes for ASCII-only strings).  So the whole yamlutil module may be irrelevant on Python 3.  Perhaps its definitions could be made conditional on Python version so it's more clear that the whole thing can go when Python 2 support is dropped.

Also maybe these assertions would be simplified by using `assertIsInstance(..., str)`?  I guess `future.builtins.str.__instancecheck__` does a thing...